packages/shared/src/archive-storybook/index.ts

```diff
@@ -1,17 +1,20 @@
 import { execFileSync } from 'child_process';
 import { resolve, dirname } from 'path';
 import { checkArchivesDirExists } from '../utils/filePaths';
+import { addViewportsToStoriesFiles } from './viewports';
 
 export function archiveStorybook(processArgs: string[], configDir: string) {
   checkArchivesDirExists();
-
-  const binPath = resolve(dirname(require.resolve('storybook/package.json')), './index.js');
-  execFileSync('node', [binPath, 'dev', ...processArgs, '-c', configDir], { stdio: 'inherit' });
+  addViewportsToStoriesFiles().then(() => {
```

Comment:
Do we need to ensure this operation is idempotent, so it doesn't keep adding viewports every time you run dev?

Reply:
@tmeasday it is idempotent because it will just overwrite the existing `modes` and `viewport` keys. I was thinking we could add a check and bail early if this has already been done, but can revisit if that seems necessary.

Reply:
Either way I'd love to see a test for it so it doesn't regress in the future!